packages/twenty-server/src/engine/core-modules/external-event/services/external-event.service.ts

```diff
@@ -0,0 +1,94 @@
+import { Injectable } from '@nestjs/common';
+
+import { ClickHouseService } from 'src/database/clickHouse/clickHouse.service';
+import { TwentyConfigService } from 'src/engine/core-modules/twenty-config/twenty-config.service';
+
+import { EventMetadataService } from './event-metadata.service';
+
+/**
+ * Interface for external event input, matching the ClickHouse schema
+ */
+export interface ExternalEventInput {
+  // The event name/type (maps to 'event' column in ClickHouse)
+  event: string;
+
+  // ID of the object related to this event
+  objectId: string;
+
+  // Type of the object related to this event (e.g., 'company', 'person', 'opportunity')
+  objectType?: string;
+
+  // User ID related to this event (optional)
+  userId?: string;
+
+  // Additional event properties (as JSON)
+  properties: Record<string, any>;
+}
+
+/**
+ * Service for handling external events
+ */
+@Injectable()
+export class ExternalEventService {
+  constructor(
+    private readonly clickHouseService: ClickHouseService,
+    private readonly twentyConfigService: TwentyConfigService,
+    private readonly eventMetadataService: EventMetadataService,
+  ) {}
+
+  /**
+   * Create a new external event
+   * @param workspaceId The workspace ID
+   * @param eventInput The event input data
+   * @returns Success status
```

Comment:
Remove all comments from this PR, we don't use comments (or only if it's something that the reader couldn't understand without it)